services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationEntity.kt

```diff
@@ -24,7 +19,16 @@ data class ApplicationEntity(
     var userId: String?,
     @Enumerated(EnumType.STRING) val applicationType: ApplicationType,
     @Type(type = "json") @Column(columnDefinition = "jsonb") var applicationData: ApplicationData,
+    @ManyToOne @JoinColumn(updatable = false, nullable = false) var hanke: HankeEntity,
 ) {
     fun toApplication() =
-        Application(id, alluid, alluStatus, applicationIdentifier, applicationType, applicationData)
+        Application(
+            id,
+            alluid,
+            alluStatus,
+            applicationIdentifier,
+            applicationType,
+            applicationData,
+            hanke.hankeTunnus ?: throw HankeNotFoundException(""),
```

Comment:
This should throw something else. HankeNotFoundException will show up as 404 Not Found to the user. This is clearly an internal error. The hanketunnus should be created when initially saving the hanke.